File: tests/expected/infer_ops.go
package main

import (
	"fmt"
)

func Foo() {
	var a int = 10
	var b int = 20
	var c1 int = (a + b)
	if !(c1 == 30) {
		panic("assert")
	}
	var c2 int = (a - b)
	if !(c2 == -10) {
		panic("assert")
	}
	var c3 int = (a * b)
	if !(c3 == 200) {
		panic("assert")
	}
	var c4 float64 = float64((a / b))
	if !(c4 == 0.5) {
		panic("assert")
	}
	var c5 int = -(a)
	if !(c5 == -10) {
		panic("assert")
	}
	var d float64 = 2.0
	var e1 float64 = (float64(a) + d)
	if !(e1 == 12.0) {
		panic("assert")
	}
	var e2 float64 = (float64(a) - d)
	if !(e2 == 8.0) {
		panic("assert")
	}
	var e3 float64 = (float64(a) * d)
	if !(e3 == 20.0) {
		panic("assert")
	}
	var e4 float64 = (float64(a) / d)
	if !(e4 == 5.0) {
		panic("assert")
	}
	var f float64 = -3.0
	if !(f < -2.9) {
		panic("assert")
	}
}

func Add1(x int8, y int8) int16 {
	return int16((x + y))
}

func Add2(x int16, y int16) int32 {
	return int32((x + y))
}

func Add3(x int32, y int32) int64 {
	return int64((x + y))
}

func Add4(x int64, y int64) int64 {
	return (x + y)
}

func Add5(x uint8, y uint8) uint16 {
	return uint16((x + y))
}

func Add6(x uint16, y uint16) uint32 {
	return uint32((x + y))
}

func Add7(x uint32, y uint32) uint64 {
	return uint64((x + y))
}

func Add8(x uint64, y uint64) uint64 {
	return (x + y)
}

func Add9(x int8, y uint16) uint32 {
	return uint32((uint16(x) + y))
}

func Sub(x int8, y int8) int8 {
	return (x - y)
}

func Mul(x int8, y int8) int16 {
	return int16((x * y))
}

func Fadd1(x int8, y float64) float64 {
	return (float64(x) + y)
}

func Show() {
	Foo()
	if !(Fadd1(6, 6.0) == 12) {
		panic("assert")
	}
	fmt.Printf("%v\n", "OK")
}

func main() {
	Foo()
	Show()
}

File: tests/expected/infer_ops.jl

function foo()
    a = 10
    b = 20
    c1 = (a + b)
    @assert(c1 == 30)
    c2 = (a - b)
    @assert(c2 == -10)
    c3 = (a * b)
    @assert(c3 == 200)
    c4 = (a / b)
    @assert(c4 == 0.5)
    c5 = -(a)
    @assert(c5 == -10)
    d = 2.0
    e1 = (a + d)
    @assert(e1 == 12.0)
    e2 = (a - d)
    @assert(e2 == 8.0)
    e3 = (a * d)
    @assert(e3 == 20.0)
    e4 = (a / d)
    @assert(e4 == 5.0)
    f = -3.0
    @assert(f < -2.9)
end

function add1(x::Int8, y::Int8)::Int16
    return (x + y)
end

function add2(x::Int16, y::Int16)::Int32
    return (x + y)
end

function add3(x::Int32, y::Int32)::Int64
    return (x + y)
end

function add4(x::Int64, y::Int64)::Int64
    return (x + y)
end

function add5(x::UInt8, y::UInt8)::UInt16
    return (x + y)
end

function add6(x::UInt16, y::UInt16)::UInt32
    return (x + y)
end

function add7(x::UInt32, y::UInt32)::UInt64
    return (x + y)
end

function add8(x::UInt64, y::UInt64)::UInt64
    return (x + y)
end

function add9(x::Int8, y::UInt16)::UInt32
    return (x + y)
end

function sub(x::Int8, y::Int8)::Int8
    return (x - y)
end

function mul(x::Int8, y::Int8)::Int16
    return (x * y)
end

function fadd1(x::Int8, y::Float64)::Float64
    return (x + y)
end

function show()
    foo()
    @assert(fadd1(convert(Int8, 6), 6.0) == 12)
    println(join(["OK"], " "))
end

function main()
    foo()
    show()
end

main()
